legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.java

```diff
@@ -253,8 +253,17 @@ private void sendResponse(final RestChannel channel, final String message, final
         channel.sendResponse(new BytesRestResponse(status, message));
     }
 
-    private void reportError(final RestChannel channel, final Exception e, final RestStatus status) {
-        sendResponse(channel, ErrorMessageFactory.createErrorMessage(e, status.getStatus()).toString(), status);
+    /**
+     * Report Error message to user.
+     * @param channel : Rest channel to sent response through.
+     * @param e : Exception caught when attempting query.
+     * @param status : Status for rest request made.
+     * @param newSqlEngineError : Error message for new SQL engine. Can be removed when old SQL engine is deprecated.
+     */
+    private void reportError(final RestChannel channel, final Exception e, final RestStatus status, String newSqlEngineError) {
+        sendResponse(channel, ErrorMessageFactory.createErrorMessage(e, status.getStatus()).toString() + (newSqlEngineError.isEmpty()
+            ? "" : "\nQuery failed both legacy and new SQL engines, see error message below for new SQL engine error.\n"
+            + newSqlEngineError), status);
```

Comment:
Putting a conditional within a concat is clunky.  I would suggest saving the result of the conditional to a String variable would be cleaner.